pkg/api/validate_plugin.go

```diff
@@ -0,0 +1,163 @@
+package api
+
+import (
+	"fmt"
+	"regexp"
+
+	pluginapi "github.com/openshift/openshift-azure/pkg/api/plugin/api"
+)
+
+var (
+	// Image version format check x{3}.y{4}.z{8}
+	imageVersion = regexp.MustCompile(`^[0-9]{3}.[0-9]{1,4}.[0-9]{8}$`)
+)
+
+// ValidatePluginTemplate validates an Plugin API external template/config struct
+func (v *Validator) ValidatePluginTemplate(t *pluginapi.Config) (errs []error) {
+	if t.ImageOffer != "osa" {
+		errs = append(errs, fmt.Errorf("imageOffer should be osa"))
+	}
+	if t.ImagePublisher != "redhat" {
+		errs = append(errs, fmt.Errorf("imagePublisher should be redhat"))
+	}
+	// validate valid ImageSKU's
+	switch t.ImageSKU {
+	case "osa_311":
+	default:
+		errs = append(errs, fmt.Errorf("invalid ImageSKU %q", t.ImageSKU))
+	}
+	// validate ImageVersion
+	if !imageVersion.MatchString(t.ImageVersion) {
+		errs = append(errs, fmt.Errorf("invalid ImageVersion %q", t.ImageVersion))
+	}
+	// validate geneva configuration inputs
+	if t.GenevaLoggingSector == "" {
+		errs = append(errs, fmt.Errorf("genevaLoggingSector cannot be empty"))
+	}
+	if t.GenevaLoggingAccount == "" {
+		errs = append(errs, fmt.Errorf("genevaLoggingAccount cannot be empty"))
+	}
+	if t.GenevaLoggingNamespace == "" {
+		errs = append(errs, fmt.Errorf("genevaLoggingNamespace cannot be empty"))
+	}
+	if t.GenevaLoggingControlPlaneAccount == "" {
+		errs = append(errs, fmt.Errorf("genevaLoggingControlPlaneAccount cannot be empty"))
+	}
+	if t.GenevaMetricsAccount == "" {
+		errs = append(errs, fmt.Errorf("genevaMetricsAccount cannot be empty"))
+	}
+	if t.GenevaMetricsEndpoint == "" {
+		errs = append(errs, fmt.Errorf("genevaMetricsEndpoint cannot be empty"))
+	}
+	// validate certificates
+	errs = append(errs, v.validatePluginTemplateCertificates(t.Certificates)...)
+	errs = append(errs, v.validatePluginTemplateImages(t.Images)...)
+	return errs
+}
+
+func (v *Validator) validatePluginTemplateCertificates(c pluginapi.CertificateConfig) (errs []error) {
+	if c.GenevaLogging.Key == nil {
+		errs = append(errs, fmt.Errorf("invalid GenevaLogging key"))
```

Comment:
maybe align these error messages with those in `validatePluginTemplateImages` so something like "GenevaLogging key cannot be empty".

Reply:
done